### PR TITLE
Display EvaluationView after exiting game

### PR DESCRIPTION
**Pull request by julianweise**

### CODE REVIEW

File: src/tetris/profiler/profile.ts [hunk 1]
```diff
@@ -21,6 +21,13 @@ export default class Profile {
 		return this._getProperty(FPP_FACE_ANALYSIS_KEY) as ProfileData<FppFaceAnalysis>;
 	}
 
+	public get image(): string {
+		if(!this.fppFaceAnalysis.value) {
+			return
```
Comment (Johannes-Schneider) [inline src/tetris/profiler/profile.ts]: Missing `;`.

File: src/tetris/scene/menuScene.ts [hunk 2]
```diff
@@ -81,13 +82,21 @@ export default class MenuScene extends Phaser.Scene {
...
-		this._playButton = new TextButton(this, menuStartX, 0, "blue_button00.png", "blue_button01.png", "Join Matchmaking", this._joinMatchmaking.bind(this));
-		this._optionsButton = new TextButton(this, menuStartX, 0, "blue_button00.png", "blue_button01.png", "Leave Matchmaking", this._leaveMatchmaking.bind(this));
+
+		this._playButton = new TextButton(this, buttonsXPosition, 0, "blue_button00.png", "blue_button01.png", "Join Matchmaking", this._joinMatchmaking.bind(this));
+		this._optionsButton = new TextButton(this, buttonsXPosition, 0, "blue_button00.png", "blue_button01.png", "Leave Matchmaking", this._leaveMatchmaking.bind(this));
+
 		this._playButton.y = menuStartY;
 		this._optionsButton.y = menuStartY + this._playButton.height + spacing;
+
+		if(this._game.profiler) {
```
Comment (Johannes-Schneider) [inline src/tetris/scene/menuScene.ts]: It seems pretty weird, that the game can only be exited if the `Profiler` exists...
I guess this check should be performed at a different place.

File: src/tetris/ui/dialog/evaluationView.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+		this._informationContainer = this._htmlElement.querySelector("#evaluation-view-basic-information");
+		this._pictureContainer = this._htmlElement.querySelector("#evaluation-view-picture");
+		this._timelineContainer = this._htmlElement.querySelector("#evaluation-view-timeline");
+		this._game = game;
+		this._displayTitle("Evaluate your player profile");
+	}
+	//endregion
+
+	//region private members
+	private _game: Game;
```
Comment (Johannes-Schneider) [inline src/tetris/ui/dialog/evaluationView.ts]: `_game` can be `readonly`.

File: src/tetris/ui/dialog/evaluationView.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+	private get biasEngine(): BiasEngine {
+		return this._game.biasEngine;
+	}
+	//endregion
+
+	//region private methods
+	private _displayBasicInformation(): void {
+		const nodes = [];
+		nodes.push(EvaluationView._createDataWrapper('age', this.profile.age));
```
Comment (friedow) [inline src/tetris/ui/dialog/evaluationView.ts]: Refactor using Object.keys() as discussed.

File: src/tetris/ui/dialog/evaluationView.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+		nodes.push(EvaluationView._createDataWrapper('skin-health', (this.profile.skinHealth * 100).toFixed(2) + "%"));
+		nodes.push(EvaluationView._createDataWrapper('number-of-matches', this.profile.numberOfMatches));
+		nodes.push(EvaluationView._createDataWrapper('time-played', this.profile.timePlayed / 1000));
+
+		nodes.forEach(datum => this._displayDatum(datum));
+	}
+
+	private _displayLastPlayerImage(): void {
+		if (this.profile.image) {
+			this._pictureContainer.classList.remove("hide");
```
Comment (friedow) [inline src/tetris/ui/dialog/evaluationView.ts]: Use 'display' as default class and just toggle 'hidden'.

File: src/tetris/ui/dialog/evaluationView.ts [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+			this._pictureContainer.classList.add("hide");
+		}
+	}
+
+	private static _getDisplayBiasValue(biasValue: number): string {
+		const factor = (biasValue - BiasEngine.NEUTRAL_BIAS_VALUE) / (BiasEngine.BIAS_RANGE / 2);
+		return (factor * 100).toFixed(4) + "%";
+	}
+
+	private _displayTimeline(): void {
```
Comment (friedow) [inline src/tetris/ui/dialog/evaluationView.ts]: Can you extract small functions from this function to increase readability?

File: src/tetris/ui/dialog/evaluationView.ts [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+				start: measurement.timestamp,
+				group: timelineGroups.get(measurement.dataSourceName).id,
+				title: measurement.value.printHTML()
+			});
+		});
+
+		this._timeline = new Timeline(this._timelineContainer, timelineItems, Array.from(timelineGroups.values()), {});
+	}
+
+	private _displayDatum(datum: Node): void {
```
Comment (friedow) [inline src/tetris/ui/dialog/evaluationView.ts]: `displayDatum` 😄

File: src/tetris/ui/dialog/evaluationView.ts [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	private static _createDataWrapper(key: string, value: string | number | boolean): HTMLDivElement {
+		if(!key || !value) {
+			return;
+		}
+		const wrapper = document.createElement('div');
+		wrapper.classList.add(CSS_CLASS_DATA_WRAPPER);
+		wrapper.classList.add(key);
+		wrapper.setAttribute("id", "evaluation-" + key);
```
Comment (friedow) [inline src/tetris/ui/dialog/evaluationView.ts]: Use `id` property please.

https://www.w3schools.com/jsref/prop_html_id.asp

File: src/tetris/ui/dialog/evaluationView.ts [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+		wrapper.innerHTML = "<strong>" + key.replace(/-/g, " ") + ":</strong> " + value.toString() || "undefined";
```
Comment (friedow) [inline src/tetris/ui/dialog/evaluationView.ts]: Use js template strings here too.

https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/template_strings

File: src/tetris/ui/dialog/evaluationView.ts (hunk 4 above)
Comment (julianweise) [reply src/tetris/ui/dialog/evaluationView.ts]: Discussed it again and decided: no!

File: src/tetris/ui/dialog/evaluationView.ts (hunk 7 above)
Comment (julianweise) [reply src/tetris/ui/dialog/evaluationView.ts]: 🥇

**Pull request merged**